src/RequestHandler.h

```diff
@@ -16,24 +16,24 @@ class RequestHandler {
   ///
   /// Modelled after Publisher::subscribe, hence must synchronously call
   /// Subscriber::onSubscribe, and provide a valid Subscription.
-  virtual Subscriber<Payload>& handleRequestChannel(
+  virtual std::shared_ptr<Subscriber<Payload>> handleRequestChannel(
       Payload request,
-      Subscriber<Payload>& response) = 0;
+      const std::shared_ptr<Subscriber<Payload>>& response) = 0;
 
   /// Handles a new Stream requested by the other end.
   virtual void handleRequestStream(
       Payload request,
-      Subscriber<Payload>& response) = 0;
+      const std::shared_ptr<Subscriber<Payload>>& response) = 0;
 
   /// Handles a new inbound Subscription requested by the other end.
   virtual void handleRequestSubscription(
       Payload request,
-      Subscriber<Payload>& response) = 0;
+      const std::shared_ptr<Subscriber<Payload>>& response) = 0;
```

Comment:
Nullability is incidental here—the real benefit is that if the implementor wants to store `response`, it can do so with proper memory management (refcounting).